graphics/shaderc/Portfile

```diff
@@ -0,0 +1,207 @@
+# of all of Shaderc's dependencies, which then gets #include'd by the
+# GLSL compiler (glslc). The project uses a Python script to do this,
+# but the script assumes that all of the dependencies' sources are
+# downloaded locally into the Shaderc project and get compiled along
+# with Shaderc itself. Since all of Shaderc's dependencies are already
+# available as MacPorts packages, we use the following Tcl procedure to
+# generate the file using MacPorts, instead of the Python script.
+proc generate_versions_file {} {
+    global name version prefix worksrcpath
+
+    set   versions_file [open ${worksrcpath}/build-version.inc w]
+    puts $versions_file "\"${name} v${version}\\n\""
+    foreach library [list spirv-tools glslang] {
+        set ver_num [exec ${prefix}/bin/port -q info --version $library]
+        puts $versions_file "\"$library $ver_num\\n\""
+    }
+    close $versions_file
+}
+
+post-patch {
+    # Don't allow the project to compile any third-party dependencies;
+    # all dependencies are already available as MacPorts packages
+    reinplace {/^add_subdirectory.third_party/s/^/#/} \
+        ${worksrcpath}/CMakeLists.txt
+    reinplace "/add_subdirectory.third_party/a\\
+        \\
+        find_host_package(glslang REQUIRED)\\
+        if(\${glslang_FOUND})\\
+        \\  message(STATUS \"Found glslang: TRUE\")\\
+        endif()\\
+        \\
+        find_host_package(SPIRV-Headers REQUIRED)\\
+        if(\${SPIRV-Headers_FOUND})\\
+        \\  message(STATUS \"Found SPIRV-Headers: TRUE\")\\
+        endif()\\
+        \\
+        find_host_package(SPIRV-Tools REQUIRED)\\
+        if(\${SPIRV-Tools_FOUND})\\
+        \\  message(STATUS \"Found SPIRV-Tools: TRUE\")\\
+        endif()\\
+    " \
+        ${worksrcpath}/CMakeLists.txt
+    foreach re [list \
+        "s|\${glslang_SOURCE_DIR}|${prefix}/include/glslang|" \
+        "s|\${spirv-tools_SOURCE_DIR}/include|${prefix}/include/spirv-tools|" \
+        "s|\${SPIRV-Headers_SOURCE_DIR}/include|${prefix}/include/spirv|" \
+        {s/\(glslang\).*glslang/\1/} \
+    ] {
+        reinplace -q $re \
+            ${worksrcpath}/CMakeLists.txt \
+            ${worksrcpath}/lib${name}_util/CMakeLists.txt \
+            ${worksrcpath}/lib${name}/CMakeLists.txt \
+            ${worksrcpath}/glslc/CMakeLists.txt
+    }
+
+    # Generate the build-version.inc file using our Tcl procedure,
+    # instead of the upstream project's included Python script.
+    reinplace {/add_custom_target.build-version/,/Update build-version.inc/s/^/#/} \
+        ${worksrcpath}/CMakeLists.txt
+    reinplace "/Update build-version.inc/a\\
+        add_custom_target(build-version\\
+        \\  COMMAND \${CMAKE_COMMAND} -E copy\\
+        \\          \${CMAKE_SOURCE_DIR}/build-version.inc\\
+        \\          \${CMAKE_CURRENT_BINARY_DIR}/build-version.inc\\
+        )\\
+    " \
+        ${worksrcpath}/CMakeLists.txt
+    generate_versions_file
+}
```

Comment:
much, if not all, of the reinplace stuff above and below if this comment should be done as patches; no e of this is readable and maintainable long-term

Reply:
Whether or not it's readable or maintainable is for me to decide, since this port is maintained by me, and isn't `openmaintainer`.

Reply:
not completely true; patchfiles are preferred for anything besides  a simple "reinplace". Feel free to keep it like this and find someone  else to merge this. I cannot follow what's happening with all this patching/reinplacing and other non-default things that are happening in this Portfile and will thus not merge this PR as-is.